guide/slash-commands/response-methods.md

```diff
@@ -0,0 +1,230 @@
+		await wait(4000);
+		await interaction.editReply('Pong!');
+	}
+});
+```
+
+If you have a command that performs longer tasks, be sure to call `deferReply()` as early as possible.
+
+Note that if you want your response to be ephemeral, you must pass an `ephemeral` flag to the `InteractionDeferOptions` here:
+
+<!-- eslint-skip -->
+
+```js
+await interaction.deferReply({ ephemeral: true });
+```
+
+It is not possible to edit a reply to change its ephemeral state once sent.
+
+::: tip
+If you want to make a proper ping command, one is available in our [FAQ](/popular-topics/faq.md#how-do-i-check-the-bot-s-ping).
+:::
+
+## Follow-ups
+
+The `reply()` and `deferReply()` methods are both *initial* responses, which tell Discord that your bot successfully receieved the interaction, but cannot be used to send additional messages. This is where follow-up messages come in. After having initially responded to an interaction, you can use `ChatInputCommandInteraction#followUp()` to send additional messages:
+
+::: warning
+After the initial response, an interaction token is valid for 15 minutes, so this is the timeframe in which you can edit the response and send follow-up messages.
+:::
+
+```js {6}
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isChatInputCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		await interaction.reply('Pong!');
+		await interaction.followUp('Pong again!');
+	}
+});
+```
+
+If you run this code you should end up having something that looks like this:
+
+<DiscordMessages>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction profile="user" :command="true">ping</DiscordInteraction>
+		</template>
+		Pong!
+	</DiscordMessage>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction profile="bot">Pong!</DiscordInteraction>
+		</template>
+		Pong again!
+	</DiscordMessage>
+</DiscordMessages>
+
+You can also pass an `ephemeral` flag to the `InteractionReplyOptions`:
+
+<!-- eslint-skip -->
+
+```js
+await interaction.followUp({ content: 'Pong again!', ephemeral: true });
+```
+
+<DiscordMessages>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction profile="user" :command="true">ping</DiscordInteraction>
+		</template>
+		Pong!
+	</DiscordMessage>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction profile="bot" :ephemeral="true">Pong!</DiscordInteraction>
+		</template>
+		Pong again!
+	</DiscordMessage>
+</DiscordMessages>
+
+Note that if you use `followUp()` after a `deferReply()`, the first follow-up will edit the `<application> is thinking` message rather than sending a new one.
+
+That's all, now you know everything there is to know on how to reply to slash commands! 
+
+::: tip
+Interaction responses can use masked links (e.g. `[text](http://site.com)`) and global emojis in the message content.
+:::
+
+## Fetching and deleting responses
+
+::: danger
```

Comment:
think this should be a warning, there isn't a whole lot of danger potential here